foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```
@@ -0,0 +1,283 @@
+    }
+    private boolean isTestDockerPom(Model model){
+        return model.getParent().getArtifactId().endsWith(TESTS_DOCKER_PARENT_ARTIFACT_ID) && model.getArtifactId().endsWith(TESTS_DOCKER_ARTIFACT_ID);
+    }
+
+    private Xpp3Dom getConfiguration(Plugin plugin) {
+        Object pluginConfiguration = plugin.getConfiguration();
+        return (Xpp3Dom) pluginConfiguration;
+    }
+
+    private static boolean hasConfigurationItem(Xpp3Dom configuration, String configurationItem){
+        return configuration.getChildren(configurationItem).length > 0;
+    }
+
+    /**
+     * Function to validate whether migration should be performed or not.
+     * @param file file to validate
+     * @return true if migration should be performed
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        boolean migrateConfiguration = false;
+        boolean migratePackaging = false;
+
+        Model model = getLocationAnnotatedModel(file);
+        if (isRootProjectPom(model) || isTestDockerPom(model)){
+            shouldExecute = false;
+        } else {
+            Plugin orphedomosPlugin = getPlugin(model, ORPH_ARTIFACT_ID);
+            if (orphedomosPlugin != null){
+                migrateConfiguration = true;
+            }
+            if(containsOrphedomosPackaging(file)){
+                migratePackaging = true;
+            }
+            shouldExecute = migrateConfiguration || migratePackaging;
+        }
+
+        return shouldExecute;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true
+     * @param file - file to migrate
+     * @return isSuccessful - whether file was migrated successfully or not
+     */
+    @Override
+    protected boolean performMigration(File file) {
+        Model model = getLocationAnnotatedModel(file);
+        Plugin plugin  = getPlugin(model, ORPH_ARTIFACT_ID);
```

If getPlugin() throws an exception instead of a null, then you can just wrap it all in a try catch and remove the null checking.